src/v2/models/AppState.js

```diff
@@ -328,6 +328,24 @@ export default Model.extend({
     const isCurrentAuthenticatorEmail = this.get('currentAuthenticatorEnrollment')?.key === AUTHENTICATOR_KEY.EMAIL;
 
     return isSameExceptMessages && isChallengeAuthenticator && isCurrentAuthenticatorEmail;
-  }
+  },
 
+  /**
+   * It will remove empty optional attributes from the model for attributes rendered as dropdown and text box.
+   */
+  cleanModelState(attributes) {
+    const currentViewState = this.getCurrentViewState();
+    if(currentViewState) {
+      const uiSchema = currentViewState.uiSchema;
+      _.each(attributes, (value, name) => {
+        if (_.isEmpty(value)) {
+          const uiSchemaProperty = uiSchema.find(schema => schema.name === name);
+          if (!_.isUndefined(uiSchemaProperty) && !uiSchemaProperty.required &&
+            (uiSchemaProperty.type === 'select' || uiSchemaProperty.type === 'text')) {
```

Comment:
I feel ui specific deletion should not be done.. It should be strictly based on `model`. So let's remove 
`(uiSchemaProperty.type === 'select' || uiSchemaProperty.type === 'text')`

and keep it to delete all non-required empty but defined attributes..

At least now we have a function that clearly says what it does. We can extend its functionality in future

Reply:
you can ignore above... just change the function name to `cleanViewModelState`